xaitk_saliency/impls/saliency/fsal.py

```diff
@@ -0,0 +1,65 @@
+from xaitk_saliency import ImageClassifierSaliencyMapGenerator
+from xaitk_saliency.utils.masking import weight_regions_by_scalar
+
+import numpy as np
+
+
+class Fsal (ImageClassifierSaliencyMapGenerator):
+    """
+    This saliency implementation transforms black-box
+    image classification scores into saliency heatmaps. This should
+    require a sequence of per-class confidences predicted on the
+    reference image, a number of per-class confidences as predicted
+    on perturbed images, as well as the masks of the reference image
+    perturbations (as would be output from a
+    `PerturbImage` implementation.
+    """
+
+    def generate(
+            self,
+            image_conf: np.ndarray,
+            perturbed_conf: np.ndarray,
+            perturbed_masks: np.ndarray
+    ) -> np.ndarray:
```

Comment:
We should not repeat inherited function doc-strings. Pass-through of the doc-string into the leaf implementation is automatically handled. E.g.
```python
import xaitk_saliency

class TestImpl (xaitk_saliency.ImageSimilaritySaliencyMapGenerator):
    def generate(self):
        pass
    def get_config(self):
        pass

t = Test()
help(t.generate)
# outputs parent class doc-string because we didn't override it.
```
This just helps reduce copy-paste errors if something upstream changes and we don't remember to re-copy-paste doc-strings.

Reply:
Removed repeating doc-strings. I did not know this was done automatically. Thank you for the neat example.